extensions/socket/internal.m

```diff
@@ -0,0 +1,657 @@
+#import <LuaSkin/LuaSkin.h>
+#import <CocoaAsyncSocket/GCDAsyncSocket.h>
+#import "socket.h"
+
+// Definitions
+@interface HSAsyncSocket : GCDAsyncSocket
+@property int readCallback;
+@property int writeCallback;
+@property int connectCallback;
+@property NSTimeInterval timeout;
+@property NSMutableArray* connectedSockets;
+@end
+
+// Userdata for hs.socket objects
+static const char *USERDATA_TAG = "hs.socket";
+#define getUserData(L, idx) (__bridge HSAsyncSocket *)((asyncSocketUserData *)lua_touserdata(L, idx))->asyncSocket;
+
+
+// Delegate implementation
+@implementation HSAsyncSocket
+
+- (id)init {
+    self.readCallback = LUA_NOREF;
+    self.writeCallback = LUA_NOREF;
+    self.connectCallback = LUA_NOREF;
+    self.timeout = -1;
+    self.connectedSockets = [[NSMutableArray alloc] init];
+    return [super initWithDelegate:self delegateQueue:dispatch_get_main_queue()];
+}
+
+- (void)socket:(HSAsyncSocket *)sock didConnectToHost:(NSString *)host port:(UInt16)port {
```

Comment:
Is this guaranteed to be called on the main thread by GCDASyncSocket? I am guessing it must be, otherwise you would have had failures from LuaSkin when it noticed it was being accessed from a non-main thread, but it seems like everything else here is expecting to be happening on random threads.